mlir-sycl/test/Dialect/IR/SYCL/ops.mlir

```diff
@@ -0,0 +1,172 @@
+// RUN: sycl-mlir-opt %s | sycl-mlir-opt | FileCheck %s
+// RUN: sycl-mlir-opt %s --mlir-print-op-generic | sycl-mlir-opt | FileCheck %s
+
+!sycl_id_1_ = !sycl.id<[1], (!sycl.array<[1], (memref<1xi64, 4>)>)>
+!sycl_id_2_ = !sycl.id<[2], (!sycl.array<[2], (memref<2xi64, 4>)>)>
+!sycl_id_3_ = !sycl.id<[3], (!sycl.array<[3], (memref<3xi64, 4>)>)>
+!sycl_range_1_ = !sycl.range<[1], (!sycl.array<[1], (memref<1xi64, 4>)>)>
+!sycl_range_2_ = !sycl.range<[2], (!sycl.array<[2], (memref<2xi64, 4>)>)>
+!sycl_range_3_ = !sycl.range<[3], (!sycl.array<[3], (memref<3xi64, 4>)>)>
+
+// CHECK-LABEL: test_num_work_items
+func.func @test_num_work_items() -> !sycl_range_1_ {
+  %0 = sycl.num_work_items() : () -> !sycl_range_1_
+  return %0 : !sycl_range_1_
+}
+
+// CHECK-LABEL: test_num_work_items_dim
```

Comment:
What are you checking here except for the presence of the function? Just that compilation doesn't produce any error (non-zero exit code)?

Reply:
Presence of the function after being printed and read again, yep

Reply:
This is how other dialects are testing their operations ([see](https://github.com/intel/llvm/blob/sycl-mlir/mlir/test/Dialect/Arith/ops.mlir))